internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -630,3 +639,20 @@ func flattenVaultEncryption(resp recoveryservices.Vault) interface{} {
 	encryptionMap["infrastructure_encryption_enabled"] = encryption.InfrastructureEncryption == recoveryservices.InfrastructureEncryptionStateEnabled
 	return encryptionMap
 }
+
+func resourceRecoveryServicesVaultSoftDeleteRefreshFunc(ctx context.Context, cfgsClient *backup.ResourceVaultConfigsClient, id parse.VaultId) pluginsdk.StateRefreshFunc {
```

Comment:
as this is an api bug could we open an issue on the rest specs and then link to it here? thanks

Reply:
an issue has opened on swagger https://github.com/Azure/azure-rest-api-specs/issues/21548